HTTPServer.driver/PutCriteria.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using HTTPServer.core;
+
+namespace HTTPServer.app
+{
+    public class PutCriteria : ICriteria
+    {
+        public bool ShouldRun(Request request)
+        {
+            return request.Method.Equals("PUT") && request.HttpVersion.Equals("HTTP/1.1") && !request.Uri.Equals("/");
```

Comment:
This check for the HTTP version - i know it has to be checked for on every request but is it necessary to have to put that check in every criteria seeing as how it's more a "server-wide setting"rather than something that we're allowing to vary with every request?  remember, each request is going through a pipeline.  can you think of how we might only have this version check code in 1 place but still apply it to every request?